### PR TITLE
Issue 520 spmecc

### LINKED ISSUE #520: SPMeCC
Write a script which solves algebraic eqns in current collector to compute effective resistance `R_cc`, then solves SPMe, adding the extra current collector ohmic loss to the terminal voltage

### PR DESCRIPTION
**Pull request by rtimms**
# Description

- Adds model to compute effective resistance of current collector 
- Updates how BCs are handled in finite elements
- Add new solver for algebraic only systems (basically the root finding algo used to get consistent ICs in DAE solver
- Adds comparison of 2plus1D and 1D models
- Adds 2+1D DFN model

Fixes #520 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by rtimms** [comment]
changing `html_static_path = ["_static"]` to `html_static_path = []` in `conf.py` seems to fix this (see [here](https://github.com/readthedocs/readthedocs.org/issues/1776)) @martinjrobins I don't really understand what this does or if changing it will cause another problem? also, I'm not sure why it broke in the first place?

File: pybamm/spatial_methods/finite_volume.py [hunk 1]
```diff
@@ -225,6 +263,8 @@ def definite_integral_matrix(self, domain):
         ----------
         domain : list
             The domain(s) of integration
+        vector_type : str, optional
+            Whether to return a row or column vector (defualt is row)
```
Comment (tinosulzer) [inline pybamm/spatial_methods/finite_volume.py]: this isn't an input to `definite_integral_matrix`. I'm not entirely clear on the difference between `definite_integral_vector` and `definite_integral_matrix`?

File: results/2plus1D/compare_lead_acid_1plus1D.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    )
+
+# plot terminal voltage
+t = np.linspace(0, solution.t[-1], 100)
+for i, model in enumerate(models):
+    plt.plot(times[i](t), voltages[i](t), lw=2, label=model.name)
+plt.xlabel("Time [h]", fontsize=15)
+plt.ylabel("Terminal voltage [V]", fontsize=15)
+plt.legend(fontsize=15)
+plt.show()
```
Comment (tinosulzer) [inline results/2plus1D/compare_lead_acid_1plus1D.py]: nice, thanks! did you try 2+1 lead acid?

File: pybamm/__init__.py [hunk 3]
```diff
@@ -95,6 +95,8 @@ def version(formatted=False):
     BoundaryGradient,
     Integral,
     IndefiniteIntegral,
+    DefiniteIntegralVector,
```
[the remark this answers is not in the document]
Comment (rtimms) [reply pybamm/__init__.py]: the matrix is the same, but (I think) Integral then takes the matrix vector product with something (e.g. 1) then that broadcasts the result, which is different

File: pybamm/spatial_methods/finite_volume.py (hunk 1 above)
Comment (rtimms) [reply pybamm/spatial_methods/finite_volume.py]: yeah I think I can get rid of `definite_integral_vector` here and just pass the row/column input to `definite_integral_matrix`

File: results/2plus1D/compare_lead_acid_1plus1D.py (hunk 2 above)
Comment (rtimms) [reply results/2plus1D/compare_lead_acid_1plus1D.py]: haven't tried but I can!

**Pull request merged**